### PR TITLE
Updating sap_library to use naming module

### PR DESCRIPTION
**Pull request by KimForss**
## Problem
Apply the naming convention module for SAP Library
## Solution
Implement a separate naming module to manage resource naming

## Tests
<Please provide steps to test the PR>

## Notes
<Additional comments for the PR>

### CODE REVIEW

File: deploy/terraform/bootstrap/sap_library/module.tf [hunk 1]
```diff
@@ -7,4 +7,21 @@ module "sap_library" {
   infrastructure          = var.infrastructure
   storage_account_sapbits = var.storage_account_sapbits
   storage_account_tfstate = var.storage_account_tfstate
+  naming                  = module.sap_namegenerator.naming
+
```
Comment (rukawata) [inline deploy/terraform/bootstrap/sap_library/module.tf:11]:
```suggestion
```

File: deploy/terraform/terraform-units/modules/sap_library/variables_local.tf [hunk 2]
```diff
@@ -1,67 +1,35 @@
...
-  prefix         = upper(format("%s-%s", local.environment, local.location_short))
+  region = try(local.var_infra.region, "")
+  prefix = try(local.var_infra.resource_group.name, var.naming.prefix.LIBRARY)
 
   // Resource group
   var_rg    = try(local.var_infra.resource_group, {})
   rg_exists = try(local.var_rg.is_existing, false)
   rg_arm_id = local.rg_exists ? try(local.var_rg.arm_id, "") : ""
-  rg_name   = local.rg_exists ? "" : try(local.var_rg.name, format("%s-SAP_LIBRARY", local.prefix))
+  rg_name   = try(var.infrastructure.resource_group.name, format("%s%s", local.prefix, local.resource_suffixes["library-rg"]))
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_library/variables_local.tf:23]:
```suggestion
  rg_name   = try(var.infrastructure.resource_group.name, format("%s%s", local.prefix, local.resource_suffixes.library-rg))
```

**Review by rukawata — COMMENTED** [review]
Thanks for the PR!
Please check the comments.
I think you need to rebase `kimforss-naming-module-saplibrary` against `feature/naming-module`.
module `sap_namegenerator` does not show up in this branch: 
https://github.com/Azure/sap-hana/tree/kimforss-naming-module-saplibrary/deploy/terraform/terraform-units/modules

File: deploy/terraform/bootstrap/sap_library/module.tf [hunk 3]
```diff
@@ -2,9 +2,26 @@
   Description:
   Setup sap library
 */
-module "sap_library" {
+module sap_library {
   source                  = "../../terraform-units/modules/sap_library"
   infrastructure          = var.infrastructure
   storage_account_sapbits = var.storage_account_sapbits
   storage_account_tfstate = var.storage_account_tfstate
+   naming         = module.sap_namegenerator.naming
```
[the remark this answers is not in the document]
Comment (KimForss) [reply deploy/terraform/bootstrap/sap_library/module.tf]: fixed

File: deploy/terraform/bootstrap/sap_library/module.tf [hunk 4]
```diff
@@ -2,9 +2,26 @@
...
+
 }
+
+module sap_namegenerator {
+  source      = "../../terraform-units/modules/sap_namegenerator"
+  environment = lower(try(var.infrastructure.landscape, ""))
+  location    = try(var.infrastructure.region, "")
+  random_id   = random_id.lib-random-id.hex
+
+  //These are not needed for the library
```
[the remark this answers is not in the document]
Comment (KimForss) [reply deploy/terraform/bootstrap/sap_library/module.tf:20]: for documentation :) - I removed them

File: deploy/terraform/bootstrap/sap_library/module.tf [hunk 5]
```diff
@@ -7,4 +7,12 @@ module "sap_library" {
   infrastructure          = var.infrastructure
   storage_account_sapbits = var.storage_account_sapbits
   storage_account_tfstate = var.storage_account_tfstate
+  naming                  = module.sap_namegenerator.naming
+}
+
+module "sap_namegenerator" {
+  source      = "../../terraform-units/modules/sap_namegenerator"
```
Comment (rukawata) [inline deploy/terraform/bootstrap/sap_library/module.tf:14]: I think some inputs are missing to pass to this module, hence you see error in pipeline like below:

```
Error: Missing required argument

  on ../../../../U1310-EAUS-SAP_LIBRARY/sap-hana/deploy/terraform/bootstrap/sap_library/module.tf line 13, in module "sap_namegenerator":
  13: module "sap_namegenerator" {

The argument "management_vnet_name" is required, but no definition was found.


Error: Missing required argument

  on ../../../../U1310-EAUS-SAP_LIBRARY/sap-hana/deploy/terraform/bootstrap/sap_library/module.tf line 13, in module "sap_namegenerator":
  13: module "sap_namegenerator" {

The argument "sap_sid" is required, but no definition was found.


Error: Missing required argument

  on ../../../../U1310-EAUS-SAP_LIBRARY/sap-hana/deploy/terraform/bootstrap/sap_library/module.tf line 13, in module "sap_namegenerator":
  13: module "sap_namegenerator" {

The argument "db_sid" is required, but no definition was found.


Error: Missing required argument

  on ../../../../U1310-EAUS-SAP_LIBRARY/sap-hana/deploy/terraform/bootstrap/sap_library/module.tf line 13, in module "sap_namegenerator":
  13: module "sap_namegenerator" {

The argument "sap_vnet_name" is required, but no definition was found.

```

Comment (KimForss) [reply deploy/terraform/bootstrap/sap_library/module.tf:14]: addressed by providing defaults for optional variables

File: deploy/terraform/terraform-units/modules/sap_library/variables_local.tf [hunk 6]
```diff
@@ -1,67 +1,35 @@
...
-  prefix         = upper(format("%s-%s", local.environment, local.location_short))
+  region = try(local.var_infra.region, "")
+  prefix = try(local.var_infra.resource_group.name, var.naming.prefix.LIBRARY)
 
   // Resource group
   var_rg    = try(local.var_infra.resource_group, {})
   rg_exists = try(local.var_rg.is_existing, false)
   rg_arm_id = local.rg_exists ? try(local.var_rg.arm_id, "") : ""
-  rg_name   = local.rg_exists ? "" : try(local.var_rg.name, format("%s-SAP_LIBRARY", local.prefix))
+  rg_name   = try(var.infrastructure.resource_group.name, format("%s%s", local.prefix, local.resource_suffixes.library-rg))
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_library/variables_local.tf:23]: dash and underscore:
`~ name     = "U1354-EAUS-SAP_LIBRARY" -> "U1354-EAUS_SAP-LIBRARY" # forces replacement`

in doc:
`{ENVIRONMENT}-{REGION_MAP}-SAP_LIBRARY`

**Review by rukawata — COMMENTED** [review]
The rg name seems to have a little problem :)

File: deploy/terraform/terraform-units/modules/sap_library/variables_local.tf (hunk 6 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_library/variables_local.tf:23]: fixed

**Comment by KimForss** [comment]
The pipeline will fail as we removed the keeper for the random_id

File: deploy/terraform/terraform-units/modules/sap_library/randomid.tf [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# Generates random text for boot diagnostics storage account name
+resource "random_id"  "post_fix" {
+  byte_length = 4
+}
+
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_library/randomid.tf:5]: newline.

**Review by rukawata — COMMENTED** [review]
Looks good now!
Just need to fix one more nit thing.

**Review by rukawata — APPROVED** [review]
lgtm!

**Pull request merged**